suite-native/module-accounts/src/components/AccountDetailTokenHeader.tsx

```diff
@@ -21,25 +24,25 @@ export const AccountDetailTokenHeader = ({
     tokenContract,
 }: AccountDetailTokenHeaderProps) => {
     const tokenAccount = useSelector((state: AccountsRootState) =>
-        selectEthereumAccountToken(state, accountKey, tokenContract),
+        selectEthereumAccountTokenInfo(state, accountKey, tokenContract),
+    );
+
+    const tokenSymbol = useSelector((state: AccountsRootState) =>
+        selectEthereumAccountTokenSymbol(state, accountKey, tokenContract),
     );
 
     if (!tokenAccount || !tokenAccount.balance) return null;
 
     return (
         <VStack alignItems="center" spacing="small" marginVertical="medium">
             <HStack spacing="small" flexDirection="row" alignItems="center" justifyContent="center">
-                <CryptoIcon symbol={tokenAccount.symbol} size="extraSmall" />
-                <EthereumTokenAmountFormatter
-                    value={tokenAccount?.balance}
-                    ethereumToken={tokenAccount.symbol}
-                />
+                <CryptoIcon symbol={tokenAccount.contract} size="extraSmall" />
+                <EthereumTokenAmountFormatter value={tokenAccount?.balance} symbol={tokenSymbol} />
```

Comment:
Why not pass contract to formatter and use selector inside of Formatter?

Reply:
What are the benefits?

I would prefer to keep it this way. If we would be selecting only the token symbol inside of the formatter, I would agree with you but we need also `tokenTransfer.value`. The get this we would have to create a new selector which returns single and specific tokenTransfer. But there is a catch... Token transfer has no unique ID:

```tsx
export interface TokenTransfer {
    type: string;
    from: string;
    to: string;
    contract: string;
    name: string;
    symbol: string;
    decimals: number;
    value?: string;
    multiTokenValues?: MultiTokenValue[];
}
```
The search for a specific tokenTransfer would be super complicated and unnecessary IMO.

Reply:
I don't like that much that you need to do this everywhere:
```
const tokenSymbol = useSelector((state: AccountsRootState) =>
        selectEthereumAccountTokenSymbol(state, accountKey, tokenContract),
    );
```
Downside is that then you will have to pass extra props accountKey and tokenContract. Even with that I would say I still prefer it little bit because these two props you have already available everywhere where Formatter is used and it's easier to just pass them then doing that extra select everytime you want to use Formatter.

You can always pass `tokenTranfer.value` or `tokenTranfer.balance` as prop to Formatter, I don't see any problem with that.

I don't have strong preference about this but, from DX perspective is much easier to just pass props you already have that importing whole new selector everywhere where you need to use only because you need only `symbol`. 

Another problem with my approach could be that then you will always have to use Formatter with accountKey, which decrease reusability...

Hope this summarization makes sense :D I will leave it up to you, if you want to keep your implementation I am OK with that.